precompiles/utils/src/data.rs

```diff
@@ -174,6 +174,9 @@ impl<'a> EvmDataReader<'a> {
             cursor: 0,
         })
     }
+    pub fn input(&mut self) -> &[u8] {
```

Comment:
It was required here -> https://github.com/AstarNetwork/astar-frame/blob/83ae4ad2063c9a4f0a8d10bd81f9adf726b3a0e8/precompiles/utils/src/bytes.rs#L110
to implement `read()` function of `EvmData` trait for `BoundedBytesString`.

I have now changed the Struct to
```rust
pub struct EvmDataReader<'a> {
    pub(crate) input: &'a [u8],
    cursor: usize,
}
```
so that we don't need a `input()` function now.

Reply:
It's the same problem as before though.

We're not in pure OOP, but some of the good practice principles still apply. 
To help shed some light on what I'm talking about: 
* [Open-closed principle](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle)
* [Encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming))

Reply:
Fixed: I have added the method `get_input_from_range()` on EvmDataReader so that we don't have to make internal data pub.